### PR TITLE
Closes #140 | Add Dengue Filipino

### LINKED ISSUE #140: Create dataset loader for Dengue Filipino
Dataloader name: `dengue_filipino/dengue_filipino.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?dengue_filipino

|     Dataset| dengue_filipino |
|-------------|---|
| Description | Benchmark dataset for low-resource multiclass classification, with 4,015 training, 500 testing, and 500 validation examples, each labeled as part of five classes. Each sample can be a part of multiple classes. Collected as tweets. |
| Subsets | - |
| Languages | fil, tgl |
| Tasks | Text Classification |
| License | Unknown (unknown) |
| Homepage | https://huggingface.co/datasets/dengue_filipino |
| HF URL | https://huggingface.co/datasets/dengue_filipino |
| Paper URL | https://ieeexplore.ieee.org/document/8459963 |

### PR DESCRIPTION
**Pull request by ilhamfp**
Closes #140

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+_DESCRIPTION = """\
+Benchmark dataset for low-resource multi-label classification, with 4,015 training, 500 testing, and 500 validation examples, each labeled as part of five classes. Each sample can be a part of multiple classes. Collected as tweets.
+"""
+
+_HOMEPAGE = "https://github.com/jcblaisecruz02/Filipino-Text-Benchmarks"
+
+_LICENSE = Licenses.GPL_3_0.value
+
+_SUPPORTED_TASKS = []  # TODO: What's the appropriate task?
```
Comment (ilhamfp) [inline seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:34]: Hi reviewers 👋 , need your input here. What's the appropriate task?

Looking through the existing single-text classification tasks, I don't think any of them fits.
```
# Single Text Classification
    ASPECT_BASED_SENTIMENT_ANALYSIS = "ABSA"
    EMOTION_CLASSIFICATION = "EC"
    HOAX_NEWS_CLASSIFICATION = "HNC"
    INTENT_CLASSIFICATION = "INT"
    LEGAL_CLASSIFICATION = "LC"
    RHETORIC_MODE_CLASSIFICATION = "RMC"
    SENTIMENT_ANALYSIS = "SA"
    TAX_COURT_VERDICT = "TACOS"
    TOPIC_MODELING = "TL"
```

Seems like we need to add a general text_multi_features task const?

Comment (sabilmakbar) [reply seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:34]: Hmmm, the task listed there has the schema defined for SEACrowd Dataloader. Although the multilabel features schema already exists, little to no tasks are mapped to the multilabel schema. Furthermore, since it's domain-specific classification (w/o linguistic context appropriate for this), it should lie under general classification, as you suggest.

I'll help on formulate this early next week (and link the upcoming PR to this one). Thanks for reporting, @ilhamfp!

Comment (XoL1507) [reply seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:34]: please help this issue

**Review by XoL1507 — APPROVED** [review]

File: seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py (hunk 1 above)
Comment (ilhamfp) [reply seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:34]: Thanks! @sabilmakbar
Seems like I can use `DOMAIN_KNOWLEDGE_MULTICLASSIFICATION` now?

Comment (sabilmakbar) [reply seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:34]: yup, you can now if you have pulled the changes from `master`

File: seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+_DATASETNAME = "dengue_filipino"
+
+_DESCRIPTION = """\
+Benchmark dataset for low-resource multi-label classification, with 4,015 training, 500 testing, and 500 validation examples, each labeled as part of five classes. Each sample can be a part of multiple classes. Collected as tweets.
+"""
+
+_HOMEPAGE = "https://github.com/jcblaisecruz02/Filipino-Text-Benchmarks"
+
+_LICENSE = Licenses.GPL_3_0.value
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:32]: is it expected to be a mistake or do you find in other places suggesting this dataset license is a GPL? bcs I find this dataset has unknown license

File: seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
+from typing import Dict, List, Tuple
+
+import datasets
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:7]: does the `seacrowd.utils.constants.Tasks` being used? (from formatter)

Comment (ilhamfp) [reply seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:7]: Will be using `DOMAIN_KNOWLEDGE_MULTICLASSIFICATION`

File: seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py (hunk 2 above)
Comment (ilhamfp) [reply seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:32]: The huggingface code linked [this](https://huggingface.co/datasets/dengue_filipino/blob/d81e2d1cbcf6049ecb324a872d4e3aac60b87743/dengue_filipino.py#L40) as its home page, and it has [GPL license](https://github.com/jcblaisecruz02/Filipino-Text-Benchmarks?tab=GPL-3.0-1-ov-file#readme)

**Comment by ilhamfp** [comment]
Hmm, right. Do you suggest setting it empty instead?

**Comment by sabilmakbar** [comment]
We can use unknown license in this case rather than having it emptied

> Hmm, right. Do you suggest setting it empty instead?

**Comment by ilhamfp** [comment]
Right, updated

**Comment by MJonibek** [comment]
@ilhamfp Thank you for implementing this dataloader. There is one issue with the test set, it is the same as the train set.  The reason is that dataloader from hugging face has a small bug: 
image

They forget to change the file name from train.csv to test.csv 

It would be great if we avoid this bug. So I suggest downloading from the original s3 link and then implementing dataloader for this dataset. Here is link that can help you to implement dataloader https://huggingface.co/datasets/dengue_filipino/blob/main/dengue_filipino.py

Original dataset s3 link: https://s3.us-east-2.amazonaws.com/blaisecruz.com/datasets/dengue/dengue_raw.zip

**Review by MJonibek — CHANGES_REQUESTED** [review]
@ilhamfp Thank you for implementing this dataloader. There is one issue with the test set, it is the same as the train set.  The reason is that dataloader from hugging face has a small bug: 
image

They forget to change the file name from train.csv to test.csv 

It would be great if we avoid this bug. So I suggest downloading from the original s3 link and then implementing dataloader for this dataset. Here is link that can help you to implement dataloader https://huggingface.co/datasets/dengue_filipino/blob/main/dengue_filipino.py

Original dataset s3 link: https://s3.us-east-2.amazonaws.com/blaisecruz.com/datasets/dengue/dengue_raw.zip

**Comment by sabilmakbar** [comment]
> @ilhamfp Thank you for implementing this dataloader. There is one issue with the test set, it is the same as the train set. The reason is that dataloader from hugging face has a small bug:

Wow, great observation! Thanks @MJonibek! I'll review the metadata in a moment

**Comment by ilhamfp** [comment]
Nice catch! I'll see if @sabilmakbar gets merged for now. If there's no action in ~a week, I'll implement the fix here.

**Comment by sabilmakbar** [comment]
Btw, the merge has been taken care by the HF staff. Therefore, we may continue the data implementation as of now @ilhamfp @MJonibek

File: seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      title={Intelligent Dengue Infoveillance Using Gated Recurrent Neural Learning and Cross-Label Frequencies},
+      year={2018},
+      volume={},
+      number={},
+      pages={2-7},
+      doi={10.1109/AGENTS.2018.8459963}}
+    }
+"""
+
+_LANGUAGES = ["fil", "tgl"]
```
[the remark this answers is not in the document]
Comment (ilhamfp) [reply seacrowd/sea_datasets/dengue_filipino/dengue_filipino.py:22]: Updated

**Review by MJonibek — APPROVED** [review]
@sabilmakbar Cool, checked, LGMT

**Review by sabilmakbar — COMMENTED** [review]
add `_URLS` by filling it out with HF dataset repo so that the variable listed is consistent w/ the others. The rest is fine and it's already pass the formatter test and loading test

**Comment by sabilmakbar** [comment]
> add _URLS by filling it out with HF dataset repo so that the variable listed is consistent w/ the others. The rest is fine and it's already pass the formatter test and loading test

I'm helping on this one

**Review by sabilmakbar — APPROVED** [review]
LGTM, thanks @ilhamfp and @MJonibek!
